modules/overrides/userconfigurable/client/limits.go

```diff
@@ -25,8 +27,9 @@ func (l *Limits) GetMetricsGenerator() *LimitsMetricsGenerator {
 }
 
 type LimitsMetricsGenerator struct {
-	Processors        listtomap.ListToMap `json:"processors,omitempty"`
-	DisableCollection *bool               `json:"disable_collection,omitempty"`
+	Processors         listtomap.ListToMap `json:"processors,omitempty"`
+	DisableCollection  *bool               `json:"disable_collection,omitempty"`
+	CollectionInterval *time.Duration      `json:"collection_interval,omitempty"`
```

Comment:
The json decoder expects `time.Duration` fields to be passed in nanoseconds. This means 60s is represented as 60000000000. This seemed like an acceptable trade-off and cleaner than making this a string and calling `time.ParseDuration`.

Reply:
This should be fine, we use `time.Duration` in a lot of configuration structs and it will automatically parse durations formatted like `15s` or `1m`.

Reply:
I believe that will work when unmarshaling yaml because it's handled here:
https://github.com/go-yaml/yaml/blob/v3.0.1/decode.go#L647-L653

The json decoder won't accept a non numeric value, such as "60s".  So I think here we'll either have to use nanoseconds as I pointed out, or this would need to be a string and we'd have to figure out the best place to run `time.ParseDuration`.

Reply:
Another option could be to change the JSON field to "collection_interval_seconds" and return `*l.CollectionInterval * 1000000` in `GetCollectionInterval`

Reply:
Oh I see, it's an oversight of the json decoder to not use `time.ParseDuration`. I'd prefer to implement our own unmarshaler in that case so we can use `15s` and `1m` etc. That is easier to use and consistent with configuration in the configmap.

There are articles explaining how to implement the unmarshaler interface: https://biscuit.ninja/posts/go-unmarshalling-json-into-time-duration/

Reply:
thanks @kvrhdn. I implemented the custom marshal/unmarshal and added some tests. Let me know what you think!